server/src/services/library.service.ts

```diff
@@ -366,258 +358,208 @@ export class LibraryService {
 
     const originalFileName = parse(assetPath).base;
 
-    if (!existingAssetEntity) {
-      // This asset is new to us, read it from disk
-      this.logger.debug(`Importing new asset: ${assetPath}`);
-      doImport = true;
-    } else if (stats.mtime.toISOString() !== existingAssetEntity.fileModifiedAt.toISOString()) {
-      // File modification time has changed since last time we checked, re-read from disk
-      this.logger.debug(
-        `File modification time has changed, re-importing asset: ${assetPath}. Old mtime: ${existingAssetEntity.fileModifiedAt}. New mtime: ${stats.mtime}`,
-      );
-      doRefresh = true;
-    } else if (existingAssetEntity.originalFileName !== originalFileName) {
-      // TODO: We can likely remove this check in the second half of 2024 when all assets have likely been re-imported by all users
-      this.logger.debug(
-        `Asset is missing file extension, re-importing: ${assetPath}. Current incorrect filename: ${existingAssetEntity.originalFileName}.`,
-      );
-      doRefresh = true;
-    } else if (!job.force && stats && !existingAssetEntity.isOffline) {
-      // Asset exists on disk and in db and mtime has not changed. Also, we are not forcing refresn. Therefore, do nothing
-      this.logger.debug(`Asset already exists in database and on disk, will not import: ${assetPath}`);
-    }
+    if (asset) {
+      const mtime = await this.getMtime(assetPath);
 
-    if (stats && existingAssetEntity?.isOffline) {
-      // File was previously offline but is now online
-      this.logger.debug(`Marking previously-offline asset as online: ${assetPath}`);
-      await this.assetRepository.update({ id: existingAssetEntity.id, isOffline: false });
-      doRefresh = true;
-    }
-
-    if (!doImport && !doRefresh) {
-      // If we don't import, exit here
-      return JobStatus.SKIPPED;
-    }
+      if (asset.trashReason == AssetTrashReason.DELETED) {
+        // Asset is trashed by user, don't re-import. This is to prevent re-importing assets that are manually trashed by the user
+        this.logger.debug(`Asset is previously trashed by user, won't refresh: ${assetPath}`);
+        return JobStatus.SKIPPED;
+      } else if (asset.trashReason == AssetTrashReason.OFFLINE) {
+        this.logger.debug(`Asset is previously trashed as offline, restoring from trash: ${assetPath}`);
+        await this.assetRepository.restoreAll([asset.id]);
+        return JobStatus.SUCCESS;
 
-    let assetType: AssetType;
+        // TODO: can we send an event just like the asset restore in the trash service?
+      } else {
+        if (mtime.toISOString() === asset.fileModifiedAt.toISOString()) {
+          // Asset exists on disk and in db and mtime has not changed, do nothing
+          this.logger.debug(`Asset already exists in database and on disk, will not import: ${assetPath}`);
+          return JobStatus.SKIPPED;
+        } else {
```

Comment:
You don't need this else.

Reply:
It's in there for the log statement

Reply:
You always return before, that's why you don't need the else

Reply:
Got it!